elasticsearch-transport/lib/elasticsearch/transport/transport/http/manticore.rb

```diff
@@ -43,97 +46,156 @@ module HTTP
-
-          # Should just be run once at startup
-          def build_client(options={})
-            client_options = options[:transport_options] || {}
-            client_options[:ssl] = options[:ssl] || {}
+            @options = arguments[:options] || {}
+            @logger = options[:logger]
+            @adapter = Adapter.new(logger, options)
+            @healthcheck_path = options[:healthcheck_path] || "/"
+            # TODO handle HTTPS
+            normalized_hosts = arguments[:hosts].map {|h| normalize_host(h)}
+            @pool = Manticore::Pool.new(logger, @adapter, @healthcheck_path, normalized_hosts)
+            @protocol    = options[:protocol] || DEFAULT_PROTOCOL
+            @serializer = options[:serializer] || ( options[:serializer_class] ? options[:serializer_class].new(self) : DEFAULT_SERIALIZER_CLASS.new(self) )
+            @max_retries     = options[:retry_on_failure].is_a?(Fixnum)   ? options[:retry_on_failure]   : DEFAULT_MAX_RETRIES
+            @retry_on_status = Array(options[:retry_on_status]).map { |d| d.to_i }
 
-            @manticore = ::Manticore::Client.new(client_options)
+            setup_sniffing!
           end
 
-          # Performs the request by invoking {Transport::Base#perform_request} with a block.
-          #
-          # @return [Response]
-          # @see    Transport::Base#perform_request
-          #
-          def perform_request(method, path, params={}, body=nil)
-            super do |connection, url|
-              params[:body] = __convert_to_json(body) if body
-              params = params.merge @request_options
-              case method
-              when "GET"
-                resp = connection.connection.get(url, params)
-              when "HEAD"
-                resp = connection.connection.head(url, params)
-              when "PUT"
-                resp = connection.connection.put(url, params)
-              when "POST"
-                resp = connection.connection.post(url, params)
-              when "DELETE"
-                resp = connection.connection.delete(url, params)
+          def normalize_host(host)
+            case host
+              when URI
+                host.to_s
+              when String
+                URI.parse(host).to_s
+              when Hash
+                host = host.clone
+                host[:scheme] ||= (host[:scheme] || host[:protocol] || "http").to_s
+                if host[:scheme] == 'http'
+                  URI::HTTP.build(host)
+                elsif scheme == 'https'
+                  URI::HTTPS.build(host)
+                else
+                  raise ArgumentError, "Unrecognized scheme for host #{host}"
+                end
               else
-                raise ArgumentError.new "Method #{method} not supported"
+                raise "Host parameter #{host} is not valid! Try something like 'http://localhost:9200'!"
+            end
+          end
+
+          def setup_sniffing!
+            if options[:sniffing]
```

Comment:
One other key point is that while we could use `reload_connections` here, it doesn't respond to `reload_after`, nor would that make sense for manticore. Having a separate option makes sense here in my connection. Additionally, from a code standpoint, the only thing this does is invoke the Sniffer class. Why have the `reload_connections` option mean 'use the sniffer class'? Esp. When the standard in the main Java client is otherwise?

I don't mean to belabor the point, but I thought I should add this extra piece of evidence.